### PR TITLE
Add Ruby 3 compatibility

### PR DESCRIPTION
**Pull request by archanaserver**
For reference: https://github.com/theforeman/hammer-cli-foreman-statistics/pull/1

### CODE REVIEW

**Comment by ofedoren** [comment]
Can you please rebase and include the changes from https://github.com/theforeman/hammer-cli-foreman-statistics/pull/1/files#diff-1f76a9b36d8f19e93b082b893f5cd25bb9e5ced86d7b15859d1b9736db83e593 here?

**Comment by ofedoren** [comment]
Well, looking at the test failures, I'd suggest to either update Gemfile.lock within this PR or remove it from git tracked files (preferably).

**Comment by ofedoren** [comment]
It's still there, did you try `git rm --cached Gemfile.lock`?

**Comment by archanaserver** [comment]
> It's still there, did you try `git rm --cached Gemfile.lock`?

Yes I did that. Now trying to remove it completely and then add it to check if that way it wil work.

Edit: It didn't worked even if I removed it. Should I add it back with bundle update?

**Comment by archanaserver** [comment]
@ofedoren https://github.com/theforeman/hammer-cli-foreman-statistics/actions/runs/5311845590/jobs/9615587311?pr=2 It is saying to add Gemfile.lock after bundle install. But I can't get it.

**Comment by ofedoren** [comment]
Well, this is because Gemfile.lock was previously added. After we changed Gemfile, the CI bundle tries to install new gems and update Gemfile.lock, which is being locked by bundler (--deployment option), which I guess default behaviour here.

We could either locally update the gemfile.lock and add it to git. Or we could try to open a separate commit which removes lockfile and runs `bundle config unset deployment` on the CI (after that we could remove `bundle config unset deployment` from the CI here).

Both ways sound like stupid workarounds, but that's the only what I came up with.

**Comment by archanaserver** [comment]
@ofedoren I tried both the ways but it didn't worked either. Can we add this to workflow, to remove the `Gemfile.lock` before `bundle install`? https://github.com/theforeman/hammer-cli-foreman-statistics/commit/9e1edf239487dc94635dfd044e6cdbb7bd5e4ef3
But I can see now, it didn't worked either.

**Comment by ofedoren** [comment]
@archanaserver, https://github.com/theforeman/hammer-cli-foreman-statistics/commit/9e1edf239487dc94635dfd044e6cdbb7bd5e4ef3 commit should work, you just need to remove lockfile before `Set up Ruby` step since during this step `bundle install` runs automatically.

**Comment by archanaserver** [comment]
> @archanaserver, [9e1edf2](https://github.com/theforeman/hammer-cli-foreman-statistics/commit/9e1edf239487dc94635dfd044e6cdbb7bd5e4ef3) commit should work, you just need to remove lockfile before `Set up Ruby` step since during this step `bundle install` runs automatically.

Does this change look good https://github.com/theforeman/hammer-cli-foreman-statistics/pull/2/commits/fdacbb7bba65ccb1a834a111bdd60ef540b80d9c? This time I can see different error for ruby v2 and ruby v3. For ruby v2 it's the same error that gets occured again, I'm not sure why it is happening. And for ruby v2 it's issue with rake version. 
I'm looking into it, but what's you view on it @ofedoren?

File: .github/workflows/unit_tests.yaml [hunk 1]
```diff
@@ -43,12 +43,16 @@ jobs:
...
+      - name: Remove Gemfile.lock
+        run: rm Gemfile.lock
       - name: Set up Ruby
         uses: ruby/setup-ruby@v1
         with:
           ruby-version: ${{ matrix.ruby }}
           bundler-cache: true
           working-directory: hammer-cli-foreman-statistics
+      - name: Bundle Install
+        run: bundle install
```
Comment (evgeni) [inline .github/workflows/unit_tests.yaml]: `ruby/setup-ruby` in the step above already does a `bundle install`, so you don't have to do it again (but it's also not the reason for the weird rake behaviour you asked about)

**Comment by evgeni** [comment]
Ha!

```
      - name: Get hammer-cli-foreman-statistics
        uses: actions/checkout@v2
        with:
          repository: theforeman/hammer-cli-foreman-statistics
          ref: master
          path: hammer-cli-foreman-statistics
```

change that to

```
      - name: Get hammer-cli-foreman-statistics
        uses: actions/checkout@v2
        with:
          path: hammer-cli-foreman-statistics
```

otherwise it pulls the master version, not your PR!

and then you also don't need the `rm Gemfile.lock` as it's gone in your PR

**Comment by archanaserver** [comment]
> Ha!
> 
> ```
>       - name: Get hammer-cli-foreman-statistics
>         uses: actions/checkout@v2
>         with:
>           repository: theforeman/hammer-cli-foreman-statistics
>           ref: master
>           path: hammer-cli-foreman-statistics
> ```
> 
> change that to
> 
> ```
>       - name: Get hammer-cli-foreman-statistics
>         uses: actions/checkout@v2
>         with:
>           path: hammer-cli-foreman-statistics
> ```
> 
> otherwise it pulls the master version, not your PR!
> 
> and then you also don't need the `rm Gemfile.lock` as it's gone in your PR

Yeee! it worked. Thanks @evgeni \o/

**Review by ofedoren — APPROVED** [review]
Thanks a lot, @archanaserver and @evgeni!

**Pull request merged**